pkg/runtime/runtime.go

```diff
@@ -758,6 +780,48 @@ func (a *DaprRuntime) initPubSub() error {
 	return nil
 }
 
+// Publish is an adapter method for the runtime to pre-validate publish requests
+// And then forward them to the Pub/Sub component.
+// This method is used by the HTTP and gRPC APIs.
+func (a *DaprRuntime) Publish(req *pubsub.PublishRequest) error {
+	if allowed := a.isPubSubOperationAllowed(req.Topic, a.scopedPublishings); !allowed {
+		return fmt.Errorf("topic %s is not allowed for app id %s", req.Topic, a.runtimeConfig.ID)
+	}
+	return a.pubSub.Publish(req)
+}
+
+func (a *DaprRuntime) isPubSubOperationAllowed(topic string, topicsList []string) bool {
+	inAllowedTopics := false
+
+	// first check if allowedTopics contain it
+	if len(a.allowedTopics) > 0 {
+		for _, t := range a.allowedTopics {
+			if t == topic {
+				inAllowedTopics = true
+				break
+			}
+		}
+		if !inAllowedTopics {
+			return inAllowedTopics
```

Comment:
nit - readability  : I suggest to return "false" . Logic is correct but it is confusing to return "inAllowedTopics" . or you can rename the var "notFound" or something "notInAllowedTopics" and default to true...

Reply:
nit fixed